src/components/draw/DrawDirective.js

```diff
@@ -27,889 +28,798 @@ goog.require('ga_permalink');
+                layer.url = undefined;
+              });
+            }
+            map.removeLayer(layer);
+            return;
+          }
+          scope.statusMsgId = 'draw_file_saving';
+          var kmlString = gaExportKml.create(layer,
+              map.getView().getProjection());
+          var id = layer.adminId ||
+              gaFileStorage.getFileIdFromFileUrl(layer.url);
+          gaFileStorage.save(id, kmlString,
+              'application/vnd.google-earth.kml+xml').then(function(data) {
+            scope.statusMsgId = 'draw_file_saved';
+
+            // If a file has been created we set the correct id to the layer
+            if (data.adminId && data.adminId != layer.adminId) {
+              layer.adminId = data.adminId;
+              layer.url = data.fileUrl;
+              layer.id = 'KML||' + layer.url;
+            }
+          });
+        };
+        var saveDebounced = gaDebounce.debounce(save, 2000, false, false);
+
+
+        var dereg = [
           $rootScope.$on('$translateChangeEnd', function() {
             if (layer) {
               layer.label = $translate.instant('draw_layer_label');
             }
-          });
+          })
+        ];
 
-          // Remove interactions on destroy
-          scope.$on('$destroy', function() {
-            deactivate();
+        // Remove interactions on destroy
+        scope.$on('$destroy', function() {
+          deactivate();
+          dereg.forEach(function(item) {
+            item();
           });
+        });
 
 
-          ////////////////////////////////////
-          // Utils functions
-          ////////////////////////////////////
-          // Change cursor style on mouse move, only on desktop
-          var updateCursorStyle = function(evt) {
-            var featureFound = map.forEachFeatureAtPixel(evt.pixel,
-                featureFilter, this, layerFilter);
-            var isSketchFeature = !!featureFound && !featureFound.getStyle();
-            map.getTarget().style.cursor = (featureFound) ?
-                ((isSketchFeature) ? 'move' : 'pointer') : '';
-          };
-          var updateCursorStyleDebounced = gaDebounce.debounce(
-              updateCursorStyle, 10, false, false);
-
-          // Fix position of dropdown
-          element.find('.dropdown-toggle').click(function() {
-            var bt = $(this);
-            var dropdown = bt.next('.dropdown-menu');
-            var dropDownTop = bt.offset().top + bt.outerHeight() -
-                50; // 50 seems to be the size of the #header
-            dropdown.css('top', dropDownTop + 'px');
-            dropdown.css('left', bt.offset().left -
-                (dropdown.outerWidth() - bt.outerWidth()) + 'px');
-          });
-        }
-      };
+        ////////////////////////////////////
+        // Utils functions
+        ////////////////////////////////////
+        // Change cursor style on mouse move, only on desktop
+        var updateCursorStyle = function(evt) {
```

Comment:
Function could be renamed. It's a little bit more than just Cursor Style adaptions.